### PR TITLE
FEATURE: ONE-3352 Add method getFeatureFlags and getColorPaletteWithGid

### PR DESCRIPTION
**Pull request by ivanmjartan**

### CODE REVIEW

File: src/project.ts [hunk 1]
```diff
@@ -120,6 +120,39 @@ export class ProjectModule {
...
+    /**
+     * Fetches a chart color palette for a project represented by the given
+     * projectId parameter.
+     *
+     * @method getColorPalette
+     * @param {String} projectId - A project identifier
+     * @return {Array<IPaletteItem>} An array of obejcts representing a project's
+     * color palette
+     */
+    public getFullColorPalette(projectId: string):Promise<Array<IColorPaletteItem>> {
```
Comment (martinmosko) [inline src/project.ts]: `Array<IColorPaletteItem>>` I would expect here `IColorPaletteItem[]`

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/92/ : SUCCESS in 50s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/107/ : FAILURE in 3m 43s

File: src/project.ts [hunk 2]
```diff
@@ -120,6 +120,39 @@ export class ProjectModule {
...
+        return this.xhr.get(`/gdc/projects/${projectId}/styleSettings`)
+            .then(r => r.getData())
```
Comment (martinmosko) [inline src/project.ts]: please use our ApiResponse object :) 

As inspiration :) 
https://github.com/gooddata/gooddata-js/blob/master/src/metadata.ts#L63

File: src/project.ts (hunk 1 above)
Comment (ivanmjartan) [reply src/project.ts]: Thx my VS somehow disable validation this kind of problems are there more give me sec I will fix all of them ...

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/93/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/108/ : SUCCESS in 46s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/94/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/109/ : SUCCESS in 35s

File: src/apiResponsesInterfaces.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// (C) 2007-2017 GoodData Corporation
```
Comment (martinmosko) [inline src/apiResponsesInterfaces.ts]: 2018 ;)

Comment (ivanmjartan) [reply src/apiResponsesInterfaces.ts]: done :-)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/95/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/110/ : SUCCESS in 57s

File: src/project.ts [hunk 4]
```diff
@@ -226,4 +256,41 @@ export class ProjectModule {
...
+                    throw new ApiResponseError(
+                        apiResponse.response.statusText,
+                        apiResponse.response,
+                        apiResponse.responseBody
+                    );
+                }
+                return apiResponse.getData();
+            })
+            .then((result: IFeatureFlagsResponse) => {
+                if (result && result.featureFlags) {
```
Comment (martinmosko) [inline src/project.ts]: are those checks necessary ? as based on interface type those attributes are not optional so we can count on them :)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/97/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/112/ : SUCCESS in 56s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/98/ : SUCCESS in 40s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/113/ : SUCCESS in 2m 06s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-opensource-repository-check/99/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/114/ : SUCCESS in 37s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-merge-compliance-check/34/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/115/ : SUCCESS in 54s

**Pull request merged**